src/ui/src/app/profile/profile.component.html

```diff
@@ -0,0 +1,281 @@
+                            Changes</button>
+                    </div>
+                </form>
+            </div>
+
+        </div>
+    </div>
+</div>
+
+
+<div class="modal fade" id="contact" tabindex="-1" aria-labelledby="contactdetails" aria-hidden="true">
+    <div class="modal-dialog modal-dialog-centered modal-dialog-scrollable">
+        <div class="modal-content">
+            <div class="modal-header">
+                <h1 class="modal-title fs-5" id="contactdetails">Contact Details</h1>
+                <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
+            </div>
+            <div class="modal-body">
+                <form [formGroup]="userForm" (ngSubmit)="onSubmit()">
+                    <div class="mb-3">
+                        <label for="phoneNumber" class="form-label">Phone Number</label>
+                        <input type="text" class="form-control" id="phoneNumber" formControlName="phoneNumber">
+
+                        @if(userForm.get('phoneNumber')?.invalid && (userForm.get('phoneNumber')?.touched ||
+                        userForm.get('phoneNumber')?.dirty)) {
+                        <div class="text-danger">
+                            Valid Phone Number is required.
+                        </div>
+                        }
+
+                    </div>
+
+                    <div class="mb-3">
+                        <label for="email" class="form-label">Email</label>
+                        <input type="text" class="form-control" id="email" formControlName="email">
+                        @if(userForm.get('email')?.invalid && (userForm.get('email')?.touched ||
+                        userForm.get('email')?.dirty)) {
+                        <div class="text-danger">
+                            Valid Email is required.
+                        </div>
+                        }
+                    </div>
+
+                    <div class="mb-3">
+                        <label for="address" class="form-label">Address</label>
+                        <input type="text" class="form-control" id="address" formControlName="address">
+                        @if(userForm.get('address')?.invalid && (userForm.get('address')?.touched ||
+                        userForm.get('address')?.dirty)) {
+                        <div class="text-danger">
+                            Address is required.
+                        </div>
+                        }
+                    </div>
+
+                    <div class="mb-3">
+                        <label for="country" class="form-label">Country</label>
+                        <input type="text-box" class="form-control" id="country" formControlName="country">
+                        @if(userForm.get('country')?.invalid && (userForm.get('country')?.touched ||
+                        userForm.get('country')?.dirty)) {
```

Comment:
why you checked touched or dirty logic will the invalid logic is enough to show the message